Fix of occ config commands (#3593)
owncloud/docs: modules/admin_manual/pages/configuration/server/occ_commands/core_commands/config_commands.adoc

Config provisioning_api of app appname could not be deleted because it did not exist
----

== config:app:get

[source,console,subs="attributes+"]
----
{occ-command-example-prefix} config:app:get [options] [--] <app> <name>
----

=== Arguments

[width="100%",cols="20%,70%",]
|===
| `app`  |  Name of the app.
| `name` |  Name of the config to get.
|===

=== Options

[width="100%",cols="33%,70%",]
|===
| `--default-value[=DEFAULT-VALUE]` | If no default value is set and the config does not exist,
the command will exit with 1.
| `--output=[OUTPUT]` | The output format to use (`plain`, `json` or `json_pretty`, default is `plain`).
|===

=== Examples

[source,console,subs="attributes+"]
----
{occ-command-example-prefix} config:app:get activity installed_version
2.2.1
----

== config:app:set

[source,console,subs="attributes+"]
----
{occ-command-example-prefix} config:app:set [options] [--] <app> <name>
----

=== Arguments

[width="100%",cols="20%,70%",]
|===
| `app`  |  Name of the app.
| `name` |  Name of the config to set.
|===

=== Options

[width="100%",cols="20%,70%",]
|===
| `--value=[VALUE]`   | The new value of the config.

> | `--update-only`     | Only updates the value, if it is not set before, it is not being added.

Reviewer: ```suggestion
| `--update-only`     | Only updates the value. If it is not set before, it is not being added.
```

NOTE: While it is possible to update/set/delete the versions and installation statuses of apps and ownCloud itself, it is *not* recommended to do this directly. 
Use the `occ app:enable`, `occ app:disable` and `occ update` commands instead.

== config:list

> The `config:list` command lists all configuration values, both for your ownCloud setup, along with any apps.

Reviewer: ```suggestion
The `config:list` command lists all configuration values for your ownCloud setup as well as for any apps.
```

[source,console,subs="attributes+"]
----
{occ-command-example-prefix} config:list [options] [--] [<app>]
----

=== Arguments

[width="100%",cols="20%,70%",]
|===
| `app`
| Name of the app. You can use "_system_" to get the config.php values, or "_all_" (the default) for all apps and system.
|===

=== Options

[width="100%",cols="20%,70%",]
|===
| `--private`
| Use this option when you want to include sensitive configs, like passwords and salts.
|===

By default, passwords and other sensitive data are omitted from the report so that the output can be posted publicly (e.g., as part of a bug report). You can see a sample output in the example below.

[source,json]
----
include::{partialsdir}configuration/server/occ_command/config-list-report.json[]
----

=== Displaying Sensitive Information

To generate a full report which includes sensitive values, such as passwords and salts, use the `--private` option, as in the following example.

[source,console,subs="attributes+"]
----
{occ-command-example-prefix} config:list --private
----

=== Filtering Information Reported

The output can be filtered to just the core information, core and apps, or one specific app.

> In the example below, you can see how to filter in each of these ways.

Reviewer: ```suggestion
In the example below, you can see how to filter for each of these categories.
```

----

=== Arguments

[width="100%",cols="20%,70%",]
|===
| `name` | Name of the config to get. Specify multiple for array parameter.
|===

=== Options

[width="100%",cols="33%,70%",]
|===
| `--default-value[=DEFAULT-VALUE]` | If no default value is set and the config does not exist,
the command will exit with 1.
| `--output=[OUTPUT]`               | The output format to use (`plain`, `json` or `json_pretty`, default is `plain`).
|===

=== Examples:

[source,console,subs="attributes+"]
----
{occ-command-example-prefix} config:system:get version
10.7.0.4
----

== config:system:set

[source,console,subs="attributes+"]
----
{occ-command-example-prefix} config:system:set [options] [--] <name> (<name>)...
----

=== Arguments

[width="100%",cols="20%,70%",]
|===
| `name` |  Name of the config parameter, specify multiple for array parameter.
|===

=== Options

[width="100%",cols="20%,70%",]
|===
| `--type=[TYPE]`     | Value type to use (`string`, `integer`, `double`, `boolean`, `json`, default is `string`). +
Note: you must use json to write multi array values.
| `--value=[VALUE]`   | The new value of the config.

> | `--update-only`     | Only updates the value, if it is not set before, it is not being added.

Reviewer: ```suggestion
| `--update-only`     | Only updates the value. If it is not set before, it is not being added.
```

| `--output=[OUTPUT]` | The output format to use (`plain`, `json` or `json_pretty`, default is `plain`).
|===

> NOTE: In order to write a boolean, float, JSON, or integer value to the configuration file, you need to specify the type on your command. This applies only to the `config:system:set` command. Please see table above for available types.

Reviewer: ```suggestion
NOTE: In order to write a boolean, float, JSON, or integer value to the configuration file, you need to specify the type of your command. This applies only to the `config:system:set` command. See table above for available types.
```
